wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/NewField.kt

```diff
@@ -102,14 +102,17 @@ class Field private constructor(
     isRedacted = options.optionMatches(".*\\.redacted", "true")
   }
 
-  fun validate(linker: Linker) {
+  fun validate(linker: Linker, syntaxRules: SyntaxRules) {
     val linker = linker.withContext(this)
     if (isPacked && !isPackable(linker, type!!)) {
       linker.addError("packed=true not permitted on $type")
     }
     if (isExtension && isRequired) {
       linker.addError("extension fields cannot be required")
     }
+    if (!syntaxRules.allowUserDefinedDefaultValue() && default != null) {
```

Comment:
```suggestion
    if (default != null && !syntaxRules.allowUserDefinedDefaultValue()) {
```

No point in check the rules if the default isn't set.